test/testsuite/FPCatSearchExt.c

```diff
@@ -168,9 +168,269 @@ STATIC void test227()
     exit_test("FPCatSearchExt:test227: Catalog search");
 }
 
+/* ------------------------- */
+/* test529: Catalog search with concurrent renames
+ *
+ * Tests FPCatSearchExt robustness when items are renamed during search iteration.
+ * This validates dircache coherency under catalog search stress, important for
+ * multi-user environments where catalog search is actively used.
+ *
+ * Scenario:
+ * 1. Client 1 creates multiple files/dirs with searchable metadata (specific finder types)
+ * 2. Client 1 starts FPCatSearchExt to find items matching criteria
+ * 3. Client 2 renames some matching items while search in progress
+ * 4. Client 1 continues iterating through search results
+ * 5. Verify: No crashes, no duplicates, results consistent despite renames
+ */
+STATIC void test529()
+{
+    char base_name[256];
+    char renamed_name[256];
+    char *dir_name = "t529_catsearch_test";
+    uint16_t vol = VolID;
+    uint16_t vol2 = 0;
+    uint16_t f_bitmap;
+    uint16_t d_bitmap;
+    uint32_t rbitmap;
+    int dir = 0;
+    int i;
+    unsigned int ret;
+    char pos[16];
+    uint32_t match_count = 0;
+    uint32_t temp;
+    const DSI *dsi;
+    struct afp_filedir_parms filedir;
+    struct afp_filedir_parms filedir2;
+    int iteration = 0;
+    ENTER_TEST
+
+    if (!Quiet) {
+        fprintf(stdout,
+                "FPCatSearchExt:test529: Catalog search with concurrent renames\n");
+    }
+
+    if (Conn->afp_version < 30) {
+        test_skipped(T_AFP3);
+        goto test_exit;
+    }
+
+    if (Path[0] == '\0') {
+        test_skipped(T_PATH);
+        goto test_exit;
+    }
+
+    if (!Conn2) {
+        test_skipped(T_CONN2);
+        goto test_exit;
+    }
+
+    dsi = &Conn->dsi;
+    f_bitmap = (1 << FILPBIT_LNAME) | (1 << FILPBIT_FNUM) | (1 << FILPBIT_FINFO);
+    d_bitmap = (1 << DIRPBIT_LNAME) | (1 << DIRPBIT_DID);
+
+    /************************************
+     * Step 1: Client 1 creates directory with searchable items
+     ************************************/
+    if (!Quiet) {
+        fprintf(stdout,
+                "\t  Step 1: Client 1 creates directory with 20 searchable files\n");
+    }
+
+    dir = FPCreateDir(Conn, vol, DIRDID_ROOT, dir_name);
+
+    if (!dir) {
+        test_nottested();
+        goto test_exit;
+    }
+
+    // Create 20 files with PDF finder info (searchable by PDINFO)
```

Comment:
nitpick: we introduce a whole lot of C++ style code comments below, which looks a little messy to my eyes ^^;